python-client/giskard/datasets/base/__init__.py

```diff
@@ -315,39 +312,39 @@ def _infer_column_types(self, cat_columns: List[str]):
         Returns:
             dict: A dictionary that maps column names to their inferred types, one of 'text', 'numeric', or 'category'.
         """
-        column_types = {}
-        nuniques = self.df.nunique()
-        df_columns = list(self.df.columns)
+        _column_types = column_types.copy()
```

Comment:
Maybe we can just shadow at the beginning and keep a single `column_types`:
```py
column_types = column_types.copy()
```
Just to avoid possible future mistakes if one forgets to always use `_column_types` in the following.